### PR TITLE
feat(Alert): Added Alert component

### PR DESCRIPTION
**Pull request by nicmell**
### IDP-2129: Added alert component
Screenshot 2024-11-26 alle 09 22 43

IMPORTANT] PR Checklist

- [x] I am aware of standards and conventions adopted in this repository, defined in the [CONTRIBUTING.md file](https://github.com/mia-platform/design-system/blob/main/CONTRIBUTING.md)

#### PR conventions

Please make sure your PR complies with the following rules before submitting it.

- [x] PR title follows the `<type>(<scope>): <subject>` structure
- [x] The PR has been labeled according to the type of changes (e.g. enhancement, new component, bug).

    > **NOTE**  
    > Some labels are used to generate release note entries: you can find the complete mapping between PR labels and release note categories [here](https://github.com/mia-platform/design-system/blob/main/.github/release.yml).  
    For a more detailed overview of PR labels, please refer to the [dedicated CONTRIBUTING section](https://github.com/mia-platform/design-system/blob/main/CONTRIBUTING.md#pull-request-labels).

#### Additional code checks

Based on your changes, some of these checks may not apply. Please make sure to check the relevant items in the list.

- [x] Changes are covered by tests 
- [x] Changes to components are accessible and documented in the Storybook
- [x] Typings have been updated
- [x] New components are exported from the `src/index.ts` file
- [x] New files include the Apache 2.0 License disclaimer
- [x] The browser console does not contain errors

### CODE REVIEW

File: src/components/Alert/Alert.module.css [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  border-left-width: 40px;
+  padding: var(--spacing-padding-lg, 16px);
+  gap: var(--spacing-gap-lg, 16px);
+  align-items: center;
+
+  background-color: var(--palette-purple-100, #EDE7FF);
+  border-color: var(--palette-purple-500, #986DF1);
+
+  & > svg {
+    position: absolute;
```
Comment (Danielecina) [inline src/components/Alert/Alert.module.css:31]: can be done in another way?

File: src/components/Alert/Alert.module.css [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+.description {
+  font-size: var(--typography-bodyS-fontSize, 14px);
+  line-height: var(--typography-bodyS-lineHeight, 16px);
+  font-weight: var(--typography-bodyS-fontWeight, 400);
+  color: var(--typography-bodyS-color, #636363);
+}
+
+.content {
```
Comment (Danielecina) [inline src/components/Alert/Alert.module.css:96]: can this be explained with a comment on this choice?

File: src/components/Alert/props.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   */
+  description?: ReactNode
+
+  /**
+   * Custom children content to be rendered inside the alert.
+   */
+  children?: ReactNode
+
+  /**
+   * Indicates if the alert is closable.
```
Comment (malta895) [inline src/components/Alert/props.ts:42]: I would specify the default here

File: src/components/Alert/Alert.module.css (hunk 2 above)
Comment (nicmell) [reply src/components/Alert/Alert.module.css:96]: The problem i noticed is the typography component in the design-system uses a lot of !important in their classes, so it becomes hard to modify the style from the extern (it shouldn't be the case anyway, but for nested components can be a problem, ie. the CheckboxLabel > Checkbox > CheckboxGroup, where the style was the same of the BodyS component but the lineheight was different causing problems in spacings).

Additionally, using the css this way you don't have to write additional codes in the component, like:
if (isCompressed) {
  titleComponent = BodyS
} else {
  titleComponent = BodyM
}

Comment (nicmell) [reply src/components/Alert/Alert.module.css:96]: EDIT: i've seen the libary exposes also a component BodyX so the switch could be on the prop "size" instead, but using the css imo has always the benefit to limit the coding to the logic, and not to the styling

File: src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import MiExclamationCircle from '../../assets/icons/ExclamationCircle.svg'
+import { Type } from './types.ts'
+import XCircle from '../../assets/icons/XCircle.svg'
+import styles from './Alert.module.css'
+
+export const defaults = {
+  isClosable: true,
+}
+
+const getIcon = (type?: Type): IconComponent | undefined => {
```
[the remark this answers is not in the document]
Comment (nicmell) [reply src/components/Alert/Alert.tsx]: my bad. resolved

File: src/components/Alert/Alert.module.css (hunk 1 above)
Comment (nicmell) [reply src/components/Alert/Alert.module.css:31]: @Danielecina i've just made some tries but seems quite hard. the problem is that the container is a flex and hence i'm not able to set the iconContainer to be the full height of the alert with "height" or any other property (except for some kind of absolute positioning)

File: src/components/Alert/props.ts (hunk 3 above)
Comment (malta895) [reply src/components/Alert/props.ts:42]: Ok, so even more so, I would explain it in the JSDoc, do you agree?

Comment (nicmell) [reply src/components/Alert/props.ts]: done

**Review by Danielecina — APPROVED** [review]

**Pull request merged**